svc/PrefService.js

```diff
@@ -94,11 +94,11 @@ export class PrefService {
         if (isEqual(oldValue, value)) return;
 
         // Change local value and fire.
-        this._data[key].value = value;
-        this.fireEvent('prefChange', {key, value, oldValue});
+        this._data[key].value = cloneDeep(value);
```

Comment:
+ Would it make sense to "freeze" the right hand side here (after cloning) and also where we load the data from server and local storage below?    Could add a utility method to set a key value to a frozen object?

+ Seems like the second clone on 98 should not be needed (and we'll feel especially confident about that if we use freeze).

+ If we freeze, we could consider *not* cloning what comes out of get() and leaving that to applications.  That would be my preference, but that would be an API change....

Reply:
* Yes I think this is a great place to use freeze
* I do another clone on what is passed in the event params because I would consider this to be the same as a call to PrefService.get(), and whoever is subscribing would potentially want to grab that reference and start modifying the same way they would with the value returned from get()
* I think it makes sense from an API perspective that PrefService has an immutable store of preference values. My use case where I ran into this is probably not too common (especially considering this issue wasn't caught before now) where my pref is a json object where the keys are user-entered values. Usually when I have a json pref the structure is known, and I end up destructuring when getting the values out of the pref. Arrays may be a bit different, would expect a lot of the time to want code like `this.values = XH.getPref('myArrayPref');`, though often the array prop would be `@observable.ref` on a model and treated as immutable anyways. Perhaps we just provide a getClone() method on PrefService for convenience for those special cases?

Reply:
Got it!  Re: the second clone().  Yes, makes sense, given the current expectation that apps get a clone made for them in get.

But having said that, what would people think of having PrefService just return frozen things?  Seems like a poster child for when you want to enforce immutability and freeze, and if we want our applications to start working in this way, what better way to start?